manifests/init.pp

```diff
@@ -142,12 +150,29 @@
     order   => '01',
   }
 
+  $cron_ensure_real = $enabled ? {
+    false   => absent,
+    default => present,
+  }
+
+  $proxy_url_real = $proxy_url ? {
+    undef   => undef,
+    default => [ "http_proxy=${proxy_url}", "https_proxy=${proxy_url}"],
+  }
+
+  cron::job { 'apt-mirror':
```

Comment:
What benefit does this have over the core `cron` resource?

https://puppet.com/docs/puppet/latest/cheatsheet_core_types.html#cron

Reply:
The biggest disadvantage in comparisson to `cron::job` is that the built-in `cron` resource does not support removing environment variables, see https://puppet.com/docs/puppet/5.5/types/cron.html#cron-attribute-environment
That means, that we cannot remove the proxy var, if we don't need it on our mirror any more.


The built-in `cron` configures all cron jobs of all modules in `/var/spool/cron/crontabs/root`, whereas `cron::job` creates independent files from a template beneath `/etc/cron.d/`, where you can set specific envs for each job.

Reply:
I'm not a fan of relying on `cron::job` but the limitations make sense to me.